### PR TITLE
docs: fix empty bnf files

### LINKED ISSUE #70503: docgen: Some BNF are empty and cause empty diagrams
These BNF are empty:

alter_table_partition_by
create_table_partition_by
rename_column
show_constraints
show_index
show_keys
show_transaction

The empty diagrams are:

alter_table_partition_by
rename_column

See https://github.com/cockroachdb/generated-diagrams/issues/1

### PR DESCRIPTION
**Pull request by RichardJCai**
Clean up empty bnf files. Small renames for simplicity.

Release note: None

Resolves https://github.com/cockroachdb/cockroach/issues/70503

### CODE REVIEW

**Review by ericharmeling — CHANGES_REQUESTED** [review]
Thanks for opening this! I have a few comments/suggestions below. LMK what you think.

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-:-MlCACYi7Nd4vipRASED:b-s7pnf2)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RichardJCai and @stbof)

___
*[docs/generated/sql/bnf/create_table_partition_by.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC2IiFCgjHhWDUwazY:-MlC2IiFCgjHhWDUwazZ:banqaf7) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/docs/generated/sql/bnf/create_table_partition_by.bnf#L1)):*
> ```
> create_table_stmt ::=
> ```

We actually don't include the diagram for this BNF in the docs, so we can delete this file (and its stmtStruct in `diagrams.go`).
___
*[docs/generated/sql/bnf/show_constraints.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC2Z16AGys9x5-CtT-:-MlC2Z16AGys9x5-CtT0:b-1a4yqo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/show_constraints.bnf#L1)):*
> ```
> show_stmt ::=
> ```

We can probably delete this file now?
___
*[docs/generated/sql/bnf/show_index.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlBzZK58Fbx3Ww2iOpk:-MlBzZK58Fbx3Ww2iOpl:b-o5wmzc) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/docs/generated/sql/bnf/show_index.bnf#L1)):*
> ```
> show_indexes_stmt ::=
> ```

We actually don't include the diagram for this BNF in the docs, so we can delete this file (and its stmtStruct in `diagrams.go`).

We only include the diagram for `show_indexes.bnf`.
___
*[docs/generated/sql/bnf/show_transaction.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC074w-cmMkOicG73h:-MlC074w-cmMkOicG73i:bi0gnl4) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/show_transaction.bnf#L1)):*
> ```
> show_stmt ::=
> ```

I think we can delete this file now?
___
*[pkg/cmd/docgen/diagrams.go, line 850 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC39eS0oXcXBHgzZHv:-MlC39eT3z-wVP1eNEZU:b-245y7l) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L850)):*
<details><summary><i alt="summary">Quoted 8 lines of code…</i></summary>

> ```
> 	{
> 		name:    "create_table_partition_by",
> 		stmt:    "create_table_stmt",
> 		inline:  []string{"opt_partition_by_table", "partition_by_table"},
> 		replace: map[string]string{"opt_table_elem_list": "table_definition", "opt_interleave": ""},
> 		match:   []*regexp.Regexp{regexp.MustCompile("PARTITION")},
> 		unlink:  []string{"table_definition"},
> 	},
> ```
</details>


We actually don't include the diagram for the BNF for this struct, so we can delete this.
___
*[pkg/cmd/docgen/diagrams.go, line 1036 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC65s65IwH1Hvvga0l:-MlC65s65IwH1Hvvga0m:b-869oqy) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1036)):*
> ```Go
> 	},
> 	{
> 		name:    "alter_rename_table_stmt",
> ```

I think we want to keep `rename_column` as the name for this struct, since we need a `rename_column` diagram. 

Also, I think `alter_rename_table_stmt` is the wrong parent statement (should be `alter_table_cmd`). 

Also, the regex in the `match` property will never find `'RENAME' ('COLUMN'|name)"` in that statement (or any statement for that matter... SMH). 

So the struct should probably look like this:

```
	{
		name:  "rename_column",
		stmt:  "alter_table_cmd",
		match: []*regexp.Regexp{regexp.MustCompile("'RENAME' opt_column column_name 'TO' column_name")},
	},
```
___
*[pkg/cmd/docgen/diagrams.go, line 1309 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlBzUws-cjaD577QAcr:-MlBzUws-cjaD577QAcs:b-a0oihy) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1309)):*
<details><summary><i alt="summary">Quoted 8 lines of code…</i></summary>

> ```
> 	{
> 		name:    "show_index",
> 		stmt:    "show_indexes_stmt",
> 		inline:  []string{"with_comment"},
> 		match:   []*regexp.Regexp{regexp.MustCompile("'SHOW' 'INDEX'")},
> 		replace: map[string]string{"var_name": "table_name"},
> 		unlink:  []string{"table_name"},
> 	},
> ```
</details>


We can actually just get rid of this. We only use the `show_indexes_stmt` diagram in the docs.
___
*[pkg/cmd/docgen/diagrams.go, line 1388 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC-B7w8_fD97jjUymD:-MlC-B7w8_fD97jjUymE:b-o3xfip) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1388)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
> 	{
> 		name:  "show_transactions_stmt",
> 		stmt:  "show_stmt",
> 		match: []*regexp.Regexp{regexp.MustCompile("'SHOW' 'TRANSACTION'")},
> 	},
> ```
</details>



I think this should be:

```
	{
		name:  "show_transactions_stmt",
		stmt:  "show_transactions_stmt",
	},
```

right?
___
*[docs/generated/sql/bnf/rename_column.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC5Rbf8EVUJozXd_4L:-MlC5Rbf8EVUJozXd_4M:ba81b5w) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/rename_column.bnf#L1)):*
> ```
> alter_rename_table_stmt ::=
> ```

See my comments above re: `rename_column`. I think we want to keep that name for the BNF.

**Comment by rickystewart** [comment]
fyi -- check out the bazel build failure in Github CI

```    ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/rename_column.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_constraints.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_indexes.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_transaction.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
```

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_BazelBuild/3536724?buildTab=log&focusLine=3&linesState=3668.3673

Make sure to remove the BNF's you'v deleted from `docs/generated/sql/bnf/BUILD.bazel`.

**Review by RichardJCai — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71095)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling and @stbof)

___
*[docs/generated/sql/bnf/create_table_partition_by.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC2IiFCgjHhWDUwazY:-MlGmPE54VrMt4JA2FnR:b-rqyfwq) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/docs/generated/sql/bnf/create_table_partition_by.bnf#L1)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

We actually don't include the diagram for this BNF in the docs, so we can delete this file (and its stmtStruct in `diagrams.go`).
</blockquote></details>

Sounds good to me, anything we don't use I would definitely want to remove.
___
*[docs/generated/sql/bnf/rename_column.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC5Rbf8EVUJozXd_4L:-MlGmzR8EWlvscy0Znnt:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/rename_column.bnf#L1)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

See my comments above re: `rename_column`. I think we want to keep that name for the BNF.
</blockquote></details>

Done.
___
*[docs/generated/sql/bnf/show_constraints.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC2Z16AGys9x5-CtT-:-MlGmU_g6BnKaQqJQC1N:br0ucqm) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/show_constraints.bnf#L1)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

We can probably delete this file now?
</blockquote></details>

Removed the diagrams.go entry for it, it will automatically get generated due to the fact that we currently generate a bnf for every top level statement but we can simply ignore it.
___
*[docs/generated/sql/bnf/show_index.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlBzZK58Fbx3Ww2iOpk:-MlGmgdq4g3V2rDzy57m:ba3dfob) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/docs/generated/sql/bnf/show_index.bnf#L1)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

We actually don't include the diagram for this BNF in the docs, so we can delete this file (and its stmtStruct in `diagrams.go`).

We only include the diagram for `show_indexes.bnf`.
</blockquote></details>

Removed
___
*[docs/generated/sql/bnf/show_transaction.bnf, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC074w-cmMkOicG73h:-MlGmk1L0cW17kknDCGO:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f345ec3e38e6b53c84ffd10829cebe3cccfad7a/docs/generated/sql/bnf/show_transaction.bnf#L1)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

I think we can delete this file now?
</blockquote></details>

Done
___
*[pkg/cmd/docgen/diagrams.go, line 850 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC39eS0oXcXBHgzZHv:-MlGmmIE4MZHq5yoC4QU:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L850)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

> ```
> 	{
> 		name:    "create_table_partition_by",
> 		stmt:    "create_table_stmt",
> 		inline:  []string{"opt_partition_by_table", "partition_by_table"},
> 		replace: map[string]string{"opt_table_elem_list": "table_definition", "opt_interleave": ""},
> 		match:   []*regexp.Regexp{regexp.MustCompile("PARTITION")},
> 		unlink:  []string{"table_definition"},
> 	},
> ```

We actually don't include the diagram for the BNF for this struct, so we can delete this.
</blockquote></details>

Done.
___
*[pkg/cmd/docgen/diagrams.go, line 1036 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC65s65IwH1Hvvga0l:-MlGmq6Y8CVAKthkh4K-:b-bu91xm) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1036)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

I think we want to keep `rename_column` as the name for this struct, since we need a `rename_column` diagram. 

Also, I think `alter_rename_table_stmt` is the wrong parent statement (should be `alter_table_cmd`). 

Also, the regex in the `match` property will never find `'RENAME' ('COLUMN'|name)"` in that statement (or any statement for that matter... SMH). 

So the struct should probably look like this:

```
	{
		name:  "rename_column",
		stmt:  "alter_table_cmd",
		match: []*regexp.Regexp{regexp.MustCompile("'RENAME' opt_column column_name 'TO' column_name")},
	},
```
</blockquote></details>

Updated
___
*[pkg/cmd/docgen/diagrams.go, line 1309 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlBzUws-cjaD577QAcr:-MlGmsgaDgDN6kGOeV72:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1309)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

> ```
> 	{
> 		name:    "show_index",
> 		stmt:    "show_indexes_stmt",
> 		inline:  []string{"with_comment"},
> 		match:   []*regexp.Regexp{regexp.MustCompile("'SHOW' 'INDEX'")},
> 		replace: map[string]string{"var_name": "table_name"},
> 		unlink:  []string{"table_name"},
> 	},
> ```

We can actually just get rid of this. We only use the `show_indexes_stmt` diagram in the docs.
</blockquote></details>

Done.
___
*[pkg/cmd/docgen/diagrams.go, line 1388 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71095#-MlC-B7w8_fD97jjUymD:-MlGmtPjEnPtwT6ceKL4:b-ivecfg) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5feb2403d87576eee8d50c7c459499c4ceee098/pkg/cmd/docgen/diagrams.go#L1388)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

> ```
> 	{
> 		name:  "show_transactions_stmt",
> 		stmt:  "show_stmt",
> 		match: []*regexp.Regexp{regexp.MustCompile("'SHOW' 'TRANSACTION'")},
> 	},
> ```


I think this should be:

```
	{
		name:  "show_transactions_stmt",
		stmt:  "show_transactions_stmt",
	},
```

right?
</blockquote></details>

If `stmt` is not specified it uses the `name`

**Comment by RichardJCai** [comment]
> fyi -- check out the bazel build failure in Github CI
> 
> ```
> 20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_constraints.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
> 20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_indexes.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
> 20:38:37     ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_transaction.bnf' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
> ```
> 
> https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_BazelBuild/3536724?buildTab=log&focusLine=3&linesState=3668.3673
> 
> Make sure to remove the BNF's you'v deleted from `docs/generated/sql/bnf/BUILD.bazel`.

Thanks for calling this out, updated.

File: pkg/cmd/docgen/diagrams.go [hunk 1]
```diff
@@ -1390,7 +1364,7 @@ var specs = []stmtSpec{
 		exclude: []*regexp.Regexp{regexp.MustCompile("'SHOW' 'EXPERIMENTAL_REPLICA'")},
 	},
 	{
-		name:  "show_transaction",
+		name:  "show_transactions_stmt",
 		stmt:  "show_stmt",
```
Comment (ericharmeling) [inline pkg/cmd/docgen/diagrams.go:1368]: The stmt here should be `show_transactions_stmt`, not `show_stmt`. See https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/parser/sql.y#L5518.

Also, I missed an inline earlier (`"opt_cluster"`).

So the stmtStruct should look like this:
```
{
		name:  "show_transactions_stmt",
		inline: []string{"opt_cluster"},
	},
```

(As you pointed out, the stmt isn't needed if the name and stmt are the same).

**Review by ericharmeling — COMMENTED** [review]
Looks good! Thanks.

Just one more comment.

**Comment by ericharmeling** [comment]
LGTM @RichardJCai 

Thanks so much for iterating on this PR.

Looks like the GH CI is failing due to the bazel build file:
```
ERROR: /go/src/github.com/cockroachdb/cockroach/docs/generated/sql/bnf/BUILD.bazel:1:8: declared output 'docs/generated/sql/bnf/show_transaction.bnf' was not created by genrule. 
```

Looks like this line needs to be removed from BUILD.bazel:
```
        "show_transaction.bnf",
```

**Review by ericharmeling — APPROVED** [review]

**Comment by ericharmeling** [comment]
Anyway we could re-run the CI tests? I really don't understand what is failing based on the build logs.

**Comment by RichardJCai** [comment]
TFTR! 
bors r=ericharmeling

**Pull request merged**